crates/net/network/src/transactions/fetcher.rs

```diff
@@ -429,9 +429,8 @@ impl<N: NetworkPrimitives> TransactionFetcher<N> {
         peers: &HashMap<PeerId, PeerMetadata<N>>,
         has_capacity_wrt_pending_pool_imports: impl Fn(usize) -> bool,
     ) {
-        let mut hashes_to_request = RequestTxHashes::default();
-        let is_session_active = |peer_id: &PeerId| peers.contains_key(peer_id);
-
+        let init_capacity_req = approx_capacity_get_pooled_transactions_req_eth68(&self.info);
+        let mut hashes_to_request = RequestTxHashes::with_capacity(init_capacity_req / 2);
```

Comment:
Added for reference, need help to determine the right number here.

Not pre-allocating enough results in multiple reallocations with copies.

On the over hand over-allocating can potentially lead to cache misses due to larger memory footprint and wasted memory.

I think slightly over-allocating is less painful than under-allocating. The cost of reallocations (especially multiple ones) tends to be higher than the cost of some unused memory capacity.

Reply:
tbh, I'm slightly lost with all these constants -.- hard to decipher what the default number here actually is

EDIT: it's 1100, this is way too much imo

I guess something like this seems more reasonable

https://github.com/paradigmxyz/reth/blob/fa340b5321dde24f13fd2f8927e0fc88116a7d47/crates/net/network/src/transactions/constants.rs#L258-L258

we could also add some metrics that track the real number

Reply:
> tbh, I'm slightly lost with all these constants -.- hard to decipher what the default number here actually is

True :D 

> we could also add some metrics that track the real number

I can do it in this PR, if you think it it's okay?

Ok, so I will make `DEFAULT_MARGINAL_COUNT_HASHES_GET_POOLED_TRANSACTIONS_REQUEST` number of pre-allocations for now.